Automattic/jetpack
# Content Options: Ensure excerpt_length is an int to prevent fatal errors

## Proposed changes:

* This PR ensures `$excerpt_length` is cast to an int if it is not already an int, to prevent fatal errors. We first check if it's numeric though, as casting a string such as 'Hello' to int will return 0, which isn't ideal for an excerpt length. If it isn't numeric, we'll use the default 55.
* One occurrence was reported internally - see p1732862467954599-slack-C034JEXD1RD - possibly due to a custom filter hooking into `excerpt_length` and returning a string.

### Other information:


## Jetpack product discussion

p1732862467954599-slack-C034JEXD1RD

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* I'm unsure how to replicate the issue in practice, but to test Content Options generally to ensure no issues are introduced you can add the `Dara` theme to a test site. Open up the customizer and select Content Options. Ensure you don't have a post excerpt set on at least one post.
* It is however to test `$excerpt_length` as a string by adding `$excerpt_length = 'Test';` underneath the `apply_filter` line.
